sponsorship_compassion/model/contracts.py

```diff
@@ -581,83 +540,76 @@ def contract_waiting(self, cr, uid, ids, context=None):
                 for line in contract.contract_line_ids:
                     sponsorship = line.sponsorship_id
                     if sponsorship.state == 'active':
-                        cr.execute(
+                        self.env.cr.execute(
                             "update recurring_contract set "
                             "activation_date = current_date,is_active = True "
                             "where id = %s", [contract.id])
             elif contract.type == 'SC':
                 # Activate directly correspondence sponsorships
-                cr.execute(
+                self.env.cr.execute(
                     "update recurring_contract set "
                     "activation_date = current_date,is_active = True "
                     "where id = %s", [contract.id])
 
-        return super(sponsorship_contract, self).contract_waiting(
-            cr, uid, ids, context)
+        return super(sponsorship_contract, self).contract_waiting()
 
     ##########################################################################
     #                             PRIVATE METHODS                            #
     ##########################################################################
-    def _on_sponsorship_finished(self, cr, uid, sponsorship_ids,
-                                 context=None):
+
+    @api.multi
+    def _on_sponsorship_finished(self, sponsorships):

```

Comment:
Why is sponsorships needed ? Self should be sufficient (a sponsorship is a recurring.contract object with type = 'S')